src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHostUserInterfaceSecurity.cs

```diff
@@ -100,25 +125,76 @@ class ConsoleHostUserInterface : System.Management.Automation.Host.PSHostUserInt
                 while (userName.Length == 0);
             }
 
-            passwordPrompt = StringUtil.Format(ConsoleHostUserInterfaceSecurityResources.PromptForCredential_Password, userName
-            );
+            passwordPrompt = StringUtil.Format(ConsoleHostUserInterfaceSecurityResources.PromptForCredential_Password, userName);
 
-            //
             // now, prompt for the password
-            //
-            WriteToConsole(passwordPrompt, true);
-            password = ReadLineAsSecureString();
-            if (password == null)
+            do
             {
-                return null;
+                WriteToConsole(passwordPrompt, true);
+                password = ReadLineAsSecureString();
+                if (password == null)
+                {
+                    return null;
+                }
             }
+            while (password.Length == 0);
 
-            WriteLineToConsole();
+            if (reenterPassword)
+            {
+                reenterPasswordPrompt = StringUtil.Format(ConsoleHostUserInterfaceSecurityResources.PromptForCredential_ReenterPassword, userName);
+                passwordMismatch = StringUtil.Format(ConsoleHostUserInterfaceSecurityResources.PromptForCredential_PasswordMismatch);
+                do
```

Comment:
I wonder if the user should be prompting multiple times for re-entering password in this loop.  It is not known if the password was incorrect the first or second time it was entered.  So it may be better to exit with error immediately if the confirmation fails, and let the user try again from the beginning.

Reply:
I have changed the behavior.

Reply:
> and let the user try again from the beginning.

What is beginning? Go to user or password re-enter?